### ISSUE TITLE
consider-using-augmented-assign false-positive on dictionaries

### ISSUE THREAD

**Issue by EugeneZelenko** [pylint-dev/pylint#10032]
### Bug description

#!/usr/bin/env python3

values = {}
accumulated_values = {}
id = 0
for r in [1, 2, 3, 4, 5]:
    id += 1
    if id == 1:
        key = str(1)
        values[key]       = r
        accumulated_values[key] = r
    else:
        if key == '1':
            accumulated_values[key]  = values[key]
        else:
            previous = str(int(key) - 1)
            accumulated_values[key]  = accumulated_values[previous] + values[key]

        key           = str(1 + int(key))
        values[key]     = r
        previous = str(int(key) - 1)
        accumulated_values[key] = accumulated_values[previous] + values[key]
print(values)
print(accumulated_values)

### Command used

pylint --enable-all-extensions --disable=all --enable=consider-using-augmented-assign consider-using-augmented-assign.py

### Pylint output

************* Module consider-using-augmented-assign
consider-using-augmented-assign.py:22:8: R6104: Use '+=' to do an augmented assign directly (consider-using-augmented-assign)

------------------------------------------------------------------
Your code has been rated at 9.47/10 (previous run: 9.47/10, +0.00)

### Expected behavior

Warning should not be reported.

### Pylint version

pylint 3.3.1
astroid 3.3.5
Python 3.12.7 (heads/main:466d8c8, Oct 15 2024, 03:59:42) [GCC 13.3.0]

### OS / Environment

Rocky 8.10

**Comment by jacobtylerwalls** [comment]
Thanks for the report. Reduced to:
```py
d = {}
d2 = {}
key = str(1)
previous = str(0)
d2[key] = d2[previous] + d[key]
```

**Comment by ekanshul** [comment]
This no longer reproduces on current `main` (pylint 4.1.0-dev0, astroid 4.2.0b5, Python 3.12.13): running the snippet from the report with the message enabled produces no output. It may be worth closing, or adding the snippet as a regression test.
